src/evaluators/BinaryExpression.js

```diff
@@ -74,7 +74,7 @@ export function getPureBinaryOperationResultType(
   } else if (op === ">>>" || op === "<<" || op === ">>" || op === "&" || op === "|" || op === "^" ||
              op === "**" || op === "%" || op === "/" || op === "*" || op === "-") {
     return reportErrorIfNotPure(IsToNumberPure, NumberValue);
-  } else if (op === "in") {
+  } else if (op === "in" || op === "instanceof") {
```

Comment:
Are plain function values simple? Are bound function values simple? If so, I think the code below would have to peek into `.$BoundTargetFunction` to figure out if this operation is safe.